### PR TITLE
Only check URL-like objects against domain filters

### PR DESCRIPTION
**Pull request by ChrisLovering**
Previously a message such as 'https://google.com hello! flask.request.method' would be
filtered due to us filtering the url shortener t.me. This commit changes to logic so
that we only check parts of the messages that matched the URL regex against our
blacklist, to avoid these false-positives.

### CODE REVIEW

**Review by Xithrius — APPROVED** [review]

File: bot/exts/filters/filtering.py [hunk 1]
```diff
@@ -478,15 +478,16 @@ async def _has_urls(self, text: str) -> Tuple[bool, Optional[str]]:
         Second return value is a reason of URL blacklisting (can be None).
         """
         text = self.clean_input(text)
-        if not URL_RE.search(text):
+        matches = URL_RE.findall(text)
+        if not matches:
```
Comment (decorator-factory) [inline bot/exts/filters/filtering.py:482]: You could simplify the code a bit like this:
```py
        domain_blacklist = self._get_filterlist_items("domain_name", allowed=False)
        for match in URL_RE.finditer(text):
            for url in domain_blacklist:
                if url.lower() in match.lower():
                    return True, self._get_filterlist_value("domain_name", url, allowed=False)["comment"]
```
Then you don't need the `if` check and a new local variable.

**Review by decorator-factory — CHANGES_REQUESTED** [review]

File: bot/exts/filters/filtering.py (hunk 1 above)
Comment (decorator-factory) [reply bot/exts/filters/filtering.py:482]: Btw, why return a flag and an `Optional[str]` when you can just return `Optional[str]`? That would also play better with type checkers.

Comment (ChrisLovering) [reply bot/exts/filters/filtering.py:482]: Refactored that an pushed it, thanks :D

It seems all of the filters are expected to return a tuple like this. So changing that would require a larger change. As far as I know, #1530 plans to change how to do filtering quite a lot.

Comment (decorator-factory) [reply bot/exts/filters/filtering.py]: oh i see

**Pull request merged**